### PR TITLE
[CSS2] change status to ED, bump version to 2.2

### PR DESCRIPTION
**Pull request by gsnedders**
See https://github.com/w3c/csswg-drafts/issues/2008#issuecomment-381088973 for the last two commits, we should decide what we want to do.

e.g., do we want the spec to say:

>  If the containing block's width depends on this element, then the resulting layout is undefined in CSS 2.2.

or: 

> 
>  If the containing block's width depends on this element, then the resulting layout is undefined in CSS 2.

If we want the former, we need to drop the last commit on this PR; if we want the latter, we should squash the last two commits together and reword the commit message.

Aside from that, I wouldn't worry too much about the changes in cover.src; we can always reword them later, it's more important to have something that claims it's a 2.2 ED to start modifying generally.

### CODE REVIEW

**Comment by frivoal** [comment]
Given that specref / bikeshed make `[[CSS2]]` link to 2.1 (and presumably will update that to link to 2.2 when 2.2 gets to REC), I think it would be better to have 2.2 nth edition generally refer to CSS2, except when specifically discussing the differences between 2.0, 2.1, and 2.2

However, to do that, this last commit seems insufficient, since the about.html section keeps using CSS2 to refer to CSS2.0, in order to explain how it supersedes it.

I suggest adding a sentence like:

> The original &lt;a href="... dated URL...">1998 Recommendation of CSS2&lt;/a> is hereafter referred to as CSS2.0 in order to distinguish it from CSS2.1 and CSS2.2, even though its original name did not carry this ".0" qualifier.

then disambiguate the relevant CSS2 references accordingly.

**Comment by gsnedders** [comment]
> However, to do that, this last commit seems insufficient, since the about.html section keeps using CSS2 to refer to CSS2.0, in order to explain how it supersedes it.

I don't think we have any precedent for calling it "2.0" (which given the full title of later revisions would imply it's "CSS level 2 revision 0").

**Comment by frivoal** [comment]
Sure, but we need to call the old REC something other than just CSS2 if we want CSS2 to refer the latest of CSS2 revision x, n<sup>th</sup> edition, for any x and any n.

I think CSS2.0 works for this purpose. It is not what the thing was initially called, but is unambiguous, and anybody who knows of the 1998 REC should easily guess that it is what we're talking about.

Although that term has no precedent in writing (that I know of), I think the phrase is used occasionally in speech.

**Comment by gsnedders** [comment]
> re: CSS 2.0 - don't make up a new old version number never before used. If need be, use "CSS2 1998" in prose to disambiguate because that's when that REC was.

Still need to go through the spec and change all instances of "CSS2" (no whitespace) to "CSS2 1998" or whatever.

> This is the right edit to also update Editors (just the two of us IIRC) and add Previous Editors (move all prior editors here, plus add Elika per f2f resolution).

Your suggestion before was to leave me out for now, FWIW, and I've done that for now.

**Comment by gsnedders** [comment]
@tantek this now uses "CSS2 (1998)" throughout wherever it means to refer to it

File: css2/cover.src [hunk 1]
```diff
@@ -78,39 +94,32 @@
...
-CSS&nbsp;2.1, and then they should only reference CSS2 for those
-features, or preferably reference such feature(s) in the respective
-CSS3 Module that includes those feature(s)).
+<p>CSS&nbsp;2.2 is derived from and is intended to replace the
+CSS&nbsp;2.1 and CSS2&nbsp;(1998)
+specifications. See <a href="changes.html">appendix C of this
+document</a> for changes from CSS&nbsp;2.1,
+and <a href="/TR/2011/REC-CSS21-2011-06-07/changes.html">appendix C of
+CSS&nbsp;2.1 for changes from CSS2&nbsp;(1998)</a> (noting that certain CSS2&nbsp;(1998) features
+were removed from CSS&nbsp;2 in CSS&nbsp2.1). All future
```
Comment (Ms2ger) [inline css2/cover.src]: Missing semicolon in the latter nbsp.

Comment (gsnedders) [reply css2/cover.src]: There's lots of entities that are missing semicolons. :( Because, y'know, it's valid SGML and software should cope. Or something.

**Pull request merged**